File: JavaStudy/TopCoder/Easy/BadExam/BadExam.java
public class BadExam {
	public double newAverage(int[] marks) {
	    // 1. find max score
	    int max = marks[0];
	    for (int num : marks) {
	        if (num > max) max = num;
	    }
	    
	    // 2. readjust
	    double[] newScores = new double[marks.length];
	    for (int i = 0; i < marks.length; i++) {
	        newScores[i] = (double) marks[i] / max * 100;
	    }
	    
	    // 3. calculate average
	    double average = 0;
	    for (double num : newScores) {
	        average += num;
	    }
	    return average / marks.length;
	}
}

File: C#Study/TopCoder/Easy/BadExam/BadExam.cs
public class BadExam
{
    public double NewAverage(int[] marks)
    {
        // 1. 최고 점수 찾기
        int max = marks[0];
        foreach (int num in marks)
        {
            if (num > max)
                max = num;
        }

        // 2. 재조정
        double[] newScores = new double[marks.Length];
        for (int i = 0; i < marks.Length; i++)
        {
            newScores[i] = (double)marks[i] / max * 100;
        }

        // 3. 평균 계산
        double average = 0;
        foreach (double num in newScores)
        {
            average += num;
        }
        return average / marks.Length;
    }
}
